iaf-aspose/src/main/java/nl/nn/adapterframework/extensions/aspose/pipe/PdfPipe.java

```diff
@@ -0,0 +1,305 @@
+package nl.nn.adapterframework.extensions.aspose.pipe;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.log4j.Logger;
+
+import nl.nn.adapterframework.configuration.ConfigurationException;
+import nl.nn.adapterframework.core.IPipeLineSession;
+import nl.nn.adapterframework.core.PipeRunException;
+import nl.nn.adapterframework.core.PipeRunResult;
+import nl.nn.adapterframework.core.PipeStartException;
+import nl.nn.adapterframework.doc.IbisDoc;
+import nl.nn.adapterframework.extensions.aspose.ConversionOption;
+import nl.nn.adapterframework.extensions.aspose.services.conv.CisConversionResult;
+import nl.nn.adapterframework.extensions.aspose.services.conv.CisConversionService;
+import nl.nn.adapterframework.extensions.aspose.services.conv.impl.AsposeLicenseLoader;
+import nl.nn.adapterframework.extensions.aspose.services.conv.impl.CisConversionServiceImpl;
+import nl.nn.adapterframework.extensions.aspose.services.conv.impl.convertors.PdfAttachmentUtil;
+import nl.nn.adapterframework.pipes.FixedForwardPipe;
+import nl.nn.adapterframework.util.XmlBuilder;
+
+/**
+ * Converts files to pdf type. This pipe has two actions convert and combine. 
+ * With combine action you can attach files into main pdf file. 
+ * @author M64D844
+ *
+ */
+public class PdfPipe extends FixedForwardPipe {
+
+	private static final Logger LOGGER = Logger.getLogger(PdfPipe.class);
+	private static final String CONVERSION_OPTION = "CONVERSION_OPTION";
+	private boolean saveSeparate = false;
+	private String tempFolder = null;
+	private String fontsDirectory;
+	private String license = null;
+	private String action = null;
+	private List<String> availableActions = Arrays.asList("combine", "convert");
+	private String mainDocumentSessionKey = "defaultMainDocumentSessionKey";
+	private String fileNameToAttachSessionKey = "defaultFileNameToAttachSessionKey";
+	protected String charset = "ISO-8859-1";
+	private int sessionkeyPostfixCounter = 0;
+	private boolean isTempDirCreated = false;
+	private AsposeLicenseLoader loader;
+	
+	@Override
+	public void configure() throws ConfigurationException {
+		super.configure();
+		if(StringUtils.isNotEmpty(tempFolder)) {
```

Comment:
I expected this code to be moved to start(). Now it is in duplicate

Reply:
This code is to check if developer configured the tempFolder correctly(exists, be a folder). I think this could stay in configure. The code snippet in start method creates the tempFolder used during conversion if user does not specify the attribute. And yes, there is a double check for tempFolder both in configure and start. Any suggestions?